code/game/objects/items/weapons/trays.dm

```diff
@@ -34,88 +35,71 @@
-	spill(user, M.loc)
+	spill(user, get_turf(M))
 
 	//Note: Added a robot check to all stun/weaken procs, beccause weakening a robot causes its active modules to bug out
 	if((user.is_clumsy()) && prob(50))              //What if he's a clown?
-		to_chat(M, "<span class='warning'>You accidentally slam yourself with the [src]!</span>")
-		if (!issilicon(M))
+		to_chat(M, SPAN_WARNING("You accidentally slam yourself with the [src]!"))
+		if(!issilicon(M))
 			M.Weaken(1)
 		user.take_organ_damage(2)
-		if(prob(50))
-			playsound(M, 'sound/items/trayhit1.ogg', 50, 1)
-			return
-		else
-			playsound(M, 'sound/items/trayhit2.ogg', 50, 1) //sound playin'
-			return //it always returns, but I feel like adding an extra return just for safety's sakes. EDIT; Oh well I won't :3
-
-	var/mob/living/carbon/human/H = M      ///////////////////////////////////// /Let's have this ready for later.
+		playsound(M, pick('sound/items/trayhit1.ogg', 'sound/items/trayhit2.ogg'), 50, TRUE)
+		return
 
+	var/mob/living/carbon/human/H = M //Let's have this ready for later.
 
-	if(!(target_zone == (BP_EYES || BP_HEAD))) //////////////hitting anything else other than the eyes
-		if(prob(33) && !issilicon(M))//robots dont bleed
+	if(!(target_zone == (BP_EYES || BP_HEAD))) // hitting anything else other than the eyes
+		if(prob(33) && !issilicon(M)) // robots dont bleed
 			src.add_blood(H)
 			var/turf/location = H.loc
-			if (istype(location, /turf/simulated))
-				location.add_blood(H)     ///Plik plik, the sound of blood
+			if(istype(location, /turf/simulated))
+				location.add_blood(H) // Plik plik, the sound of blood
 
 		M.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been attacked with [src.name] by [user.name] ([user.ckey])</font>")
 		user.attack_log += text("\[[time_stamp()]\] <font color='red'>Used the [src.name] to attack [M.name] ([M.ckey])</font>")
 		msg_admin_attack("[key_name_admin(user)] used the [src.name] to attack [key_name_admin(M)] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user),ckey_target=key_name(M))
 
 		if(prob(15))
-			if(!issilicon(M)) M.Weaken(3)
+			if(!issilicon(M))
+				M.Weaken(3)
 			M.take_organ_damage(3)
 		else
 			M.take_organ_damage(5)
-		if(prob(50))
-			playsound(M, 'sound/items/trayhit1.ogg', 50, 1)
-			for(var/mob/O in viewers(M, null))
-				O.show_message(text("<span class='danger'>[] slams [] with the tray!</span>", user, M), 1)
-			return
-		else
-			playsound(M, 'sound/items/trayhit2.ogg', 50, 1)  //we applied the damage, we played the sound, we showed the appropriate messages. Time to return and stop the proc
-			for(var/mob/O in viewers(M, null))
-				O.show_message(text("<span class='danger'>[] slams [] with the tray!</span>", user, M), 1)
-			return
+		playsound(M, pick('sound/items/trayhit1.ogg', 'sound/items/trayhit2.ogg'), 50, TRUE)
+		visible_message("\The [user] slams \the [M] with \the [src]!")
```

Comment:
Why does this directly give organ damage? 🤔 
Also maybe make it slam people in the organ they actually did.

Reply:
is that in scope

Reply:
Depends on you.